src/components/TOSAgreement/TOSAgreement.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { Stack, Typography, Box, Button } from "@mui/material";
+import { SxProps } from "@mui/system";
+
+type TOSAgreementProps = {
+  productName: string;
+  description: React.ReactNode | string;
+  children: React.ReactNode;
+  onConfirm: VoidFunction;
+
+  sx?: SxProps;
+  confirmBtnDisabled?: boolean;
+  confirmBtnLabel?: string;
+};
+
+export function TOSAgreement({
+  productName,
+  description,
+  children,
+  onConfirm,
+
+  sx,
+  confirmBtnDisabled,
+  confirmBtnLabel = "Accedi",
+}: TOSAgreementProps) {
+  const isDescriptionComponentAString = typeof description === "string";
+
+  return (
+    <Stack
+      alignItems="center"
+      justifyContent="center"
+      sx={{ py: 16, backgroundColor: "#FAFAFA", ...sx }}
```

Comment:
We should add responsive values for users who attempt to login for the first time using a mobile device (e.g: Piattaforma Notifiche as a normal citizen)
```suggestion
      sx={{ py: { xs: 4, lg: 16 }, backgroundColor: "#FAFAFA", ...sx }}
```